### DISCUSSION TITLE
Only show loading state on initial load of `FutureProvider`

### DISCUSSION THREAD

**Discussion by peterhijma** [rrousselGit/riverpod#1011] [Q&A]
I'd like to show the loading state only on the initial load of `FutureProvider`. After using something like "swipe down to refresh" I'd like the list to remain visible and update if data is ready again.

Anyone knows a non-hacky way to do this right now? Or a hacky way...

**Comment by iamarnas** [comment]
You can create your own `FutureBuilder` with `AsyncValue` and `StateNotifier`. Then you can manage your loading state how you want.
#### Simple counter example:
```dart
final counterProvider = StateNotifierProvider<Counter, AsyncValue<int>>((ref) {
  return Counter();
});

class Counter extends StateNotifier<AsyncValue<int>> {
  Counter() : super(const AsyncValue.loading()) {
    increment();
  }

  int _count = 0;

  Future<void> increment() async {
    await Future.delayed(const Duration(milliseconds: 500));
    state = AsyncValue.data(_count++);
  }
}
```
